runtime/jcl/common/mgmtmemory.cpp

```diff
@@ -132,13 +81,13 @@ Java_com_ibm_java_lang_management_internal_MemoryMXBeanImpl_getNonHeapMemoryUsag
 		UDATA *ramClassFreqUDATABlockFreeListPtr = frequentlyAccessedBlockPtr->ramClassUDATABlockFreeList;
 		UDATA *ramClassInFreqUDATABlockFreeListPtr = inFrequentlyAccessedBlockPtr->ramClassUDATABlockFreeList;
 
-		subtractUDATABlockChain(ramClassSub4gUDATABlockFreeListPtr, &used);
-		subtractUDATABlockChain(ramClassFreqUDATABlockFreeListPtr, &used);
-		subtractUDATABlockChain(ramClassInFreqUDATABlockFreeListPtr, &used);
+		VM_VMHelpers::subtractUDATABlockChain(ramClassSub4gUDATABlockFreeListPtr, (I_64 *)&used);
+		VM_VMHelpers::subtractUDATABlockChain(ramClassFreqUDATABlockFreeListPtr, (I_64 *)&used);
+		VM_VMHelpers::subtractUDATABlockChain(ramClassInFreqUDATABlockFreeListPtr, (I_64 *)&used);
```

Comment:
Please remove the casts to `I_64 *`; they are guaranteed to be redundant `jlong` and `I_64` must both be 64-bit signed integer types. Also on lines 88-90.

Reply:
For the record, I checked the definitions of `jlong` and `I_64`:
```c
typedef long long jlong;
typedef int64_t I_64;
```
GCC effectively (and indirectly) says:
```c
typedef signed long int int64_t;
```
They have the same sizes, but pointers to these types are distinct.